integration/aggregate_compat_test.go

```diff
@@ -241,16 +258,43 @@ func TestAggregateCompatStages(t *testing.T) {
 	testCases := map[string]aggregateStagesCompatTestCase{
 		"MatchAndCount": {
 			pipeline: bson.A{
+				// when $match is the first stage, pushdown is done.
 				bson.D{{"$match", bson.D{{"v", 42}}}},
 				bson.D{{"$count", "v"}},
 				bson.D{{"$sort", bson.D{{"_id", 1}}}},
 			},
+			resultPushdown: true,
+		},
+		"CountAndMatch": {
+			pipeline: bson.A{
+				// when $match is the second stage, no pushdown is done.
+				bson.D{{"$count", "v"}},
+				bson.D{{"$match", bson.D{{"v", 1}}}},
+				bson.D{{"$sort", bson.D{{"_id", 1}}}},
+			},
 		},
 	}
 
 	testAggregateStagesCompat(t, testCases)
 }
 
+func TestAggregateCompatEmptyPipeline(t *testing.T) {
+	providers := []shareddata.Provider{
+		// for testing empty pipeline use a collection with single document,
+		// because sorting will not matter.
+		shareddata.Unsets,
+	}
+
+	testCases := map[string]aggregateStagesCompatTestCase{
+		"Empty": {
+			pipeline: bson.A{},
+			noSort:   true,
```

Comment:
I'm confused by this condition. What do we test by setting `noSort` specifically? What would change if we didn't have `noSort` condition at all?

Reply:
It's because of https://github.com/FerretDB/FerretDB/blob/main/integration/aggregate_compat_test.go#L32, $sort is always added to pipeline. But we do not want it to test this condition that pipeline is empty, 
https://github.com/FerretDB/FerretDB/pull/2199/files#diff-d5be5b46004f6aaf1d90e271efea546e6a16800f6ebef0f9b67049bc8f240bfdR26